### PR TITLE
77914: Update S3 logic to add custom metadata

### PR DESCRIPTION
**Pull request by balexandr**
## Summary

Pega requested that we add the form's metadata into the S3 call (put_obect()) so they can pull it out on their end.
- Updated S3 service to use `put_object()` instead of `upload_file()`
- Add sponsor info to metadata
- Update S3 spec

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/77914

## Testing done

- Ran automated tests
- Used this to check after putting it up:
```
response = s3.bucket(bucket_name).object(File.basename(file_path)).get
puts "Metadata: #{response.metadata}"
```

### CODE REVIEW

**Review by reiting — APPROVED** [review]

**Review by michaelclement — APPROVED** [review]
lgtm

File: lib/simple_forms_api_submission/s3.rb
Comment (cloudmagic80) [inline lib/simple_forms_api_submission/s3.rb:1]: I would recommend to use a block with File.open to ensure the file is properly closed after being read, especially in the case of exceptions.

**Review by michaelclement — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]
Other than that, looks good.

File: lib/simple_forms_api_submission/s3.rb
Comment (LindseySaari) [reply lib/simple_forms_api_submission/s3.rb:1]: Has this comment been addressed? I want to make sure it is before I take a look and review

File: lib/simple_forms_api_submission/s3.rb [hunk 1]
```diff
@@ -14,13 +14,22 @@ def initialize(region:, access_key_id:, secret_access_key:, bucket_name:)
...
+        # Convert nil values to empty strings in the metadata
+        metadata&.transform_values! { |value| value || '' }
 
-      { success: true }
-    rescue => e
-      { success: false, error_message: "S3 Upload failure for #{file}: #{e.message}" }
+        client.put_object({
+                            bucket: Settings.ivc_forms.s3.bucket,
+                            key:,
+                            body: File.read(file),
```
Comment (LindseySaari) [inline lib/simple_forms_api_submission/s3.rb:25]: I think the File.open comment is referring to this line here

File: lib/simple_forms_api_submission/s3.rb
Comment (balexandr) [reply lib/simple_forms_api_submission/s3.rb:1]: Yes it has

File: lib/simple_forms_api_submission/s3.rb (hunk 1 above)
Comment (balexandr) [reply lib/simple_forms_api_submission/s3.rb:25]: Yup I changed it from a read to an open

**Review by michaelclement — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**